### PR TITLE
fix/rbf update account and pending tx after push

### PR DESCRIPTION
**Pull request by szymonlesisz**
responses from the blockbook may be delayed.
modify account utxo/balance and existing (overridden) transactions after successful tx push

### CODE REVIEW

File: packages/suite/src/reducers/wallet/transactionReducer.ts [hunk 1]
```diff
@@ -64,6 +65,12 @@ const update = (
     };
 };
 
+const replace = (draft: State, key: string, txid: string, tx: WalletAccountTransaction) => {
+    const accountTxs = initializeAccount(draft, key);
+    const index = accountTxs.findIndex(t => t && t.txid === txid);
+    accountTxs[index] = tx;
```
Comment (szymonlesisz) [inline packages/suite/src/reducers/wallet/transactionReducer.ts]: if index < 1

File: packages/suite/src/utils/wallet/accountUtils.ts [hunk 2]
```diff
@@ -441,21 +440,26 @@ export const accountSearchFn = (
...
+    tx: PrecomposedTransaction,
+    txid: string,
+    prevTxid?: string,
+) => {
+    if (tx.type !== 'final') return [];
 
-    // calculate availableBalance
-    let availableBalanceBig = new BigNumber(account.availableBalance).minus(tx.totalSpent);
+    // find utxo to replace
+    const replaceUtxo = account.utxo?.filter(u => u.txid === prevTxid) || [];
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/accountUtils.ts]: if prevTxid

File: packages/suite/src/actions/wallet/transactionActions.ts [hunk 3]
```diff
@@ -55,16 +65,67 @@ export const remove = (account: Account, txs: WalletAccountTransaction[]): Trans
...
+        }
+        return action;
+    });
+    // dispatch replace actions
+    actions.forEach(a => dispatch(a));
+
+    // balance is reduced by fee differences
+    const txToReplace = transactions.find(t => t.tx.type === 'sent' || t.tx.type === 'self');
+    const prevBaseFee = txToReplace?.tx.rbfParams?.baseFee || 0;
+    const feeDiff = prevBaseFee ? newBaseFee - prevBaseFee : 0; // 0 as fallback, balance will not be reduced
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/transactionActions.ts]: move to 107

Comment (szymonlesisz) [reply packages/suite/src/actions/wallet/transactionActions.ts]: addressed e10dafd

File: packages/suite/src/utils/wallet/accountUtils.ts (hunk 2 above)
Comment (szymonlesisz) [reply packages/suite/src/utils/wallet/accountUtils.ts]: addressed e10dafd

File: packages/suite/src/reducers/wallet/transactionReducer.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/suite/src/reducers/wallet/transactionReducer.ts]: addressed e10dafd

**Review by mroz22 — APPROVED** [review]
:+1:

**Review by keraf — APPROVED** [review]

**Comment by mroz22** [comment]
I am keeping ot open, so that it does not snatch @tsusanka's attention as it should go to the release

**Pull request merged**